### PR TITLE
[20.10] vendor update containerd client code to v1.5.10

### PR DESCRIPTION
**Pull request by thaJeztah**
Possibly more dependencies need to be updated, and instead of this we should cherry-pick.
This is just a quick check "what would it look like if we bumped the version in this branch";

Updating to containerd 1.5

Last containerd update in 20.10 is https://github.com/moby/moby/commit/1f88736871858c9e2eb71365abd059a27aaed320 (https://github.com/moby/moby/pull/41688)

- https://github.com/moby/moby/commit/ab1dd80877ebe3ae9a4bf40d5b39686a18ae360e https://github.com/moby/moby/pull/42274
- https://github.com/moby/moby/commit/5761fca80dad59217fa76007c7a6cbe75a9f7d64 https://github.com/moby/moby/pull/42274
- https://github.com/moby/moby/commit/42ef2c5036f5131fbabf7389ac7efa049cd63cd2 https://github.com/moby/moby/pull/42276
- https://github.com/moby/moby/commit/6202322b60ecbc93e683353ab180934e15cd149c https://github.com/moby/moby/pull/42254
- https://github.com/moby/moby/commit/7c1c123555744d84d04893e304c39db697189e18 https://github.com/moby/moby/pull/42249
- https://github.com/moby/moby/commit/84df737f500a7d0074e97ea60bdfc02e5fe9be9f https://github.com/moby/moby/pull/42636
- https://github.com/moby/moby/commit/4fc2d4df0319feafd9b393d8791e7a5a369d01d9 https://github.com/moby/moby/pull/42656
- https://github.com/moby/moby/commit/3d58d1381145554989d0871415937a3ab4c2b87d https://github.com/moby/moby/pull/42697
- https://github.com/moby/moby/commit/582ef29426c3690d28f0de2201c95ca46c424679 https://github.com/moby/moby/pull/42994
- https://github.com/moby/moby/commit/458b4aae197aee829075c4f0a883ae689f966f29 https://github.com/moby/moby/pull/43025

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by samuelkarp — APPROVED** [review]
LGTM since CI is passing.

File: vendor/github.com/containerd/containerd/platforms/compare.go [hunk 1]
```diff
@@ -26,103 +31,70 @@ type MatchComparer interface {
...
+}
+
+// Only returns a match comparer for a single platform
+// using default resolution logic for the platform.
+//
+// For arm/v8, will also match arm/v7, arm/v6 and arm/v5
+// For arm/v7, will also match arm/v6 and arm/v5
+// For arm/v6, will also match arm/v5
+// For amd64, will also match 386
+func Only(platform specs.Platform) MatchComparer {
```
Comment (tianon) [inline vendor/github.com/containerd/containerd/platforms/compare.go:84]: Do we have decent tests on our usage of this function?

Comment (tianon) [reply vendor/github.com/containerd/containerd/platforms/compare.go:84]: More specifically, these three usages:

- https://github.com/moby/moby/blob/16009830c2e2f3840ee73459918289f4bf54bd1d/daemon/images/image.go#L59 (in `ImageService.manifestMatchesPlatform`)
- https://github.com/moby/moby/blob/16009830c2e2f3840ee73459918289f4bf54bd1d/builder/builder-next/adapters/containerimage/pull.go#L426 (in `puller.Snapshot`)
- https://github.com/moby/moby/blob/16009830c2e2f3840ee73459918289f4bf54bd1d/daemon/images/image_builder.go#L179 (in `ImageService.pullForBuilder`)

**Review by tianon — APPROVED** [review]
The only _potentially_ unexpected change I could see in a very rough review of the vendored changes was the change to `platforms.Only` -- I'm not sure whether we have great coverage on our three uses of that function, but I think it's probably fine? (Hopefully someone else knows better.)

**Comment by AkihiroSuda** [comment]
Needs rebase

**Pull request closed without merging**